packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { createContext, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+export interface AccessTokenContextValue {
+  /**
+   * The returned accessToken wheter from the props or from the fetchToken function
+   */
+  accessToken?: string
```

Comment:
```suggestion
  /**
   * The current access token, if any.
   *
   * If the AccessTokenProvider was instantiated without a token, or the fetch token function has not completed, this property will be undefined.
   *
   * Otherwise, when re-fetching an access token, this property will remain equal to the previous access token, until the fetch token function completes again.
   */
  accessToken?: string
```